packages/core-data/src/types/common.ts

```diff
@@ -0,0 +1,55 @@
+/**
+ * The raw data representation.
+ */
+export interface RawObject {
+	/**
+	 * Data as it exists in the database.
+	 */
+	raw?: string;
+	/**
+	 * Data transformed for display.
+	 */
+	rendered?: string;
+}
+
+export type RawString = string;
+export type RawData = RawObject | RawString;
+
+export type PostStatus = 'publish' | 'future' | 'draft' | 'pending' | 'private';
+
+export type OpenOrClosed = 'open' | 'closed';
+export type ActiveOrInactive = 'active' | 'inactive';
```

Comment:
for types like these it would be good to think of the role they play. we're hard-coding the values into the type name, which, while probably understandable here and fine, is somewhat redundant and going to be awkward once `ActiveOrInactive` gains its third value, `pending`, or its fourth etc…

in fact these are statuses and the property names give us those clues: `comment_status`, `status`, `ping_status`

we don't want to prioritize code modularity or DRY with types. this is an early optimization that doesn't help in most cases.

to that end, what happens if `comment_status` gets a new option like `moderated` that `ping_status` doesn't?

it's okay to define these in the modules where they're used and to have some redundancy.

```ts
// attachment.ts
type CommentStatus = 'open' | 'closed';
type PingStatus = 'open' | 'closed';
```

if after writing things separately it burns with the fire of a thousand evils to write the same thing twice or thrice then we can consider using a shared type, but in this case I wouldn't think that's necessary.

```ts
type Availability = 'open' | 'closed';

interface Attachment … {
	…
	comment_status?: Availability;
	ping_status?: Availability;
	…
}
```

Reply:
Good call! I just updated the PR